mitmproxy.py

```diff
@@ -19,33 +21,56 @@ def binary(self):
             name = "mitmdump-osx"
         elif "win" in sys.platform:
             name = "mitmdump-win.exe"
-        return os.path.join(os.getcwd(), "utils", name)
+
+        return os.path.join(os.getcwd(), "utils", "mitm-{}".format(self.mitmproxy_version), name)
 
     def __enter__(self):
-        if self.record:
-            command = [
-                self.binary,
-                "--wfile",
-                self.path,
-                "--script",
-                " ".join(
-                    [
-                        os.path.join(self.scripts, "inject-deterministic.py")]
-                )
-            ]
+
+        if self.mitmproxy_version == "2.0.2":
+            if self.record:
+                command = [
+                    self.binary,
+                    "--wfile",
+                    self.path,
+                    "--script",
+                    " ".join(
+                        [
+                            os.path.join(self.scripts, "inject-deterministic.py")]
+                    )
+                ]
+            else:
+                command = [
+                    self.binary,
+                    "--replay-kill-extra",
+                    "--script",
+                    " ".join(
+                        [
+                            os.path.join(self.scripts, "alternate-server-replay.py"),
+                            self.path,
+                        ]
+                    ),
+                ]
+        elif self.mitmproxy_version == "4.0.4":
+            if self.record:
+                command = [
+                    self.binary,
+                    # "--scripts",
+                    # os.path.join(self.scripts, "inject-deterministic.py"),
+                    "--save-stream-file",
+                    self.path
+                ]
+            else:
+                command = [
+                    self.binary,
+                    "--scripts",
+                    os.path.join(self.scripts, "alternate-server-replay-4.0.4-no-best-match.py"),
+                    "--set",
+                    "server_replay={}".format(self.path)
+                ]
         else:
-            command = [
-                self.binary,
-                "--replay-kill-extra",
-                "--script",
-                " ".join(
-                    [
-                        os.path.join(self.scripts, "alternate-server-replay.py"),
-                        self.path,
-                    ]
-                ),
-            ]
-        self.process = subprocess.Popen(command)
+            raise Exception("Could not find the mitmproxy version")
+
+        self.process = subprocess.Popen(command, env=os.environ)
```

Comment:
Isn't the env inherited from the existing environment by default? Why is this change needed?

Reply:
you are right I will remove that